app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue

```diff
@@ -218,44 +199,78 @@ const dropDownInteraction = WidgetEditHandler.New(props.input, {
   end: () => {
     dropdownVisible.value = false
   },
+  addItem: () => {
+    dropdownVisible.value = true
+    editedValue.value = undefined
+    return true
+  },
```

Comment:
This isn't in solution to that problem. Sorry, this API really needs documentation--I hadn't written it yet because the refactor to support resumable interactions will affect this code.

The `addItem` interface allows a widget to provide a handler for the add-item intent of a child widget. The vector widget emits it when its plus button is clicked, and the selection widget handles it by opening its dropdown to choose an item. If the parent doesn't handle the event, the vector widget proceeds with its default behavior of inserting a default item.

I hadn't noted the issue you describe as a defect, but now that you bring it up I see that we should change it.
The resumable-interactions refactor will also make that easy to fix.

Reply:
Ok, I see. I'm ok with it if documentation will be added.

I already introduced idea of "accepting current edit" in drop downs: https://github.com/enso-org/enso/pull/9536/files#diff-801a0f40269674dc4f01de6d16a7afae776d144bd0ff347c4a0cb0faba5bb69aR207-R211 it could be used in your PR if you find any problems with clicks.